mitodl/mitxonline
# Adds product pages to the CMS

#### Pre-Flight checklist


#### What are the relevant tickets?

#678

#### What's this PR do?

Adds program pages to the CMS. These follow the definition of a course page, but allow you to choose a program instead. In addition to that, this allows Flexible Pricing Request Forms to be created under a ProgramPage, and should update the logic to return the underlying Program if the form is created under a ProgramPage. Also updated are the `configure_wagtail` management command and routing to support ProgramPages (but not the HomePage model).

#### How should this be manually tested?

Automated tests should pass.

To run manual tests, you will need courses set up within a program, and flexible pricing tiers/discounts set up appropriately for the _program_. 

1. Run the `configure_wagtail` command. This will create the base Program index page in the CMS.
2. Add a Program page under the "Programs" in the CMS. 
3. Fill out the form as per usual and publish. The page should be viewable. 
4. Create a Flexible Pricing Request Form under the Program page, in the same manner that you'd create one for a course. 
5. Submit a Flexible Pricing Request via the new form. Courses that are in the specified program should see the proper discount. 

#### Any background context you want to provide?

The homepage code hasn't been updated to allow you to choose a Program Page for the Featured Content section. This is by design; we don't necessarily want marketing pages for programs to be seen yet. 

The cards and things that are displayed on a Program Page are cut down - it doesn't look for start dates, eligible course runs, or provide a way to enroll directly in the program. This is also on purpose.
